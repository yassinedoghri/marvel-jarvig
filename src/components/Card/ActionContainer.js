import styled from "styled-components";

const ActionContainer = styled.div`
  display: flex;
  justify-content: space-between;
  padding: ${props => props.theme.spaces.md}em;
`;

export default ActionContainer;
