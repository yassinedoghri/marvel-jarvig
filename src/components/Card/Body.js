import Paragraph from '../Paragraph';
import {media} from "../../media";

export const Body = Paragraph.extend`
    color: ${props => props.theme.colors.light.secondary};
    display: none;
    
    ${media.tablet`
        display: block;
    `}
`;

export default Body;
