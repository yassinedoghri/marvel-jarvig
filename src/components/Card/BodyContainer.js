import styled from "styled-components";
import media from "utils/media";

const BodyContainer = styled.div`
  padding: 0;

  ${media.desktop`
    padding: 1em;
  `};
`;

export default BodyContainer;
