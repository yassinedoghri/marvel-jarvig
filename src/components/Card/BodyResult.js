import styled from "styled-components";
import media from "utils/media";

const BodyResult = styled.h3`
  font-family: Bangers, sans-serif;
  color: ${props => props.theme.colors.foreground.base};
  font-size: 4rem;
  text-align: center;
  margin: 0;
  letter-spacing: 0.1em;

  & span {
    font-family: Bangers, sans-serif;
    color: ${props => props.theme.colors.accent.base};
    text-shadow: -1px 2px 0px ${props => props.theme.colors.primary.base},
      2px -1px 0px ${props => props.theme.colors.primary.base},
      4px 4px 0 ${props => props.theme.colors.foreground.base};
  }

  ${media.tablet`
    display: block;
  `};
`;

export default BodyResult;
