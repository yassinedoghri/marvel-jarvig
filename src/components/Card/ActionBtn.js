import { Button } from "components";

const ActionBtn = Button.extend`
  width: ${props => (props.half ? "48%" : "100%")};
`;

export default ActionBtn;
