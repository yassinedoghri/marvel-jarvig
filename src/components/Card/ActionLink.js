import ActionBtn from "components/Card/ActionBtn";
import { Link } from "react-router-dom";

const ActionLink = ActionBtn.withComponent(Link);

export default ActionLink;
