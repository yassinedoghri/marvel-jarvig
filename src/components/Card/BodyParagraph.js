import { Paragraph } from "components";

const BodyParagraph = Paragraph.extend`
  color: ${props => props.theme.colors.contrastLight.secondary};
`;

export default BodyParagraph;
