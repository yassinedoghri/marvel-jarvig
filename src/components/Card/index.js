import ActionBtn from "components/Card/ActionBtn";
import ActionContainer from "components/Card/ActionContainer";
import ActionLink from "components/Card/ActionLink";
import BodyContainer from "components/Card/BodyContainer";
import BodyParagraph from "components/Card/BodyParagraph";
import BodyResult from "components/Card/BodyResult";
import Hint from "components/Card/Hint";
import HintMessage from "components/Card/HintMessage";
import Ribbon from "components/Card/Ribbon";

import Title from "components/Card/Title";
import { bounceIn, shake, tada } from "react-animations";
import styled, { keyframes } from "styled-components";
import { media } from "utils";

const tadaAnimation = keyframes`${tada}`;
const bounceInAnimation = keyframes`${bounceIn}`;
const shakeAnimation = keyframes`${shake}`;

const Card = styled.div`
  position: relative;
  max-width: 30em;
  width: 100%;
  background-color: ${props => props.theme.colors.background.base};
  border: solid ${props => props.theme.colors.foreground.base};
  border-width: 3px 4px 3px 5px;
  border-radius: 95% 4% 92% 5%/4% 95% 6% 95%;
  box-shadow: 0 3px 15px rgba(0, 0, 0, 0.2);
  padding: ${props => props.theme.spaces.md}em;
  margin: 0 auto ${props => props.theme.spaces.lg}em;
  animation: 1s
    ${props => {
      if (props.animation === "correct") return tadaAnimation;
      if (props.animation === "incorrect") return shakeAnimation;
      return bounceInAnimation;
    }};

  ${media.desktop`
      width: auto;
      margin-bottom: 0;
  `};
`;

Card.Title = Title;
Card.Ribbon = Ribbon;
Card.BodyParagraph = BodyParagraph;
Card.BodyResult = BodyResult;
Card.BodyContainer = BodyContainer;
Card.ActionBtn = ActionBtn;
Card.ActionLink = ActionLink;
Card.ActionContainer = ActionContainer;
Card.Hint = Hint;
Card.HintMessage = HintMessage;

export default Card;
