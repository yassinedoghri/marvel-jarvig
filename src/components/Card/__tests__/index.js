import Card from "components/Card";
import React from "react";
import { renderWithTheme } from "utils/testHelpers";

// TODO: Fix ActionButton tests
describe("Card Block", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(
      <Card>
        <Card.Ribbon backgroundcolor="foreground">Ribbon!</Card.Ribbon>
        [<Card.Hint key={1}>X</Card.Hint>,
        <Card.HintMessage key={2}>This is a hint!</Card.HintMessage>]
        <Card.Title>Card Title</Card.Title>
        <Card.BodyContainer>
          <Card.BodyParagraph center>Card text</Card.BodyParagraph>
        </Card.BodyContainer>
        <Card.ActionContainer>
          {/* Card Action Buttons inside */}
        </Card.ActionContainer>
      </Card>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
