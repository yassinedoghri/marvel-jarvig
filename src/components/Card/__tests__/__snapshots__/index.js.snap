// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Block renders correctly 1`] = `
<div
  className="sc-gisBJw eLWijE"
>
  <p
    className="sc-fAjcbJ gwDekQ"
  >
    Ribbon!
  </p>
  [
  <button
    className="sc-dVhcbM kwrEQA"
  >
    X
  </button>
  ,
  <div
    className="sc-eqIVtm blkeIP"
  >
    This is a hint!
  </div>
  ]
  <h2
    className="sc-caSCKo bYqcUu"
  >
    Card Title
  </h2>
  <div
    className="sc-jhAzac fhqzSw"
  >
    <p
      className="sc-fBuWsC gdYvkp"
    >
      Card text
    </p>
  </div>
  <div
    className="sc-bRBYWo bzSgXm"
  />
</div>
`;
